src/index.js

```diff
@@ -45,13 +48,213 @@ module.exports = {
       }],
     }, {
       type: 'function',
-      fn(opts) { return localRunner.run(opts); },
+      fn(opts) { 
+      // if W doesn't merge, return run(), if W merge, return statistic()
+//        console.log('opts.remoteResult.data is',opts.remoteResult.data);
+//         console.log('opts.previousData is', opts.previousData);
+       console.log('opts.remoteResult.data.endOfIteration is' , opts.remoteResult.data.endOfIteration);
+       console.log('opts.remoteResult.data.statisticStep is', opts.remoteResult.data.statisticStep);
+        
+        if ((opts.remoteResult.data.endOfIteration == true) && (opts.remoteResult.data.statisticStep == 0)){
```

Comment:
I’m a little iffy on these branches based on remote result state. Can we separate these into different local computations and let COINSTAC’s pipeline runner do the work?